app/src/main/java/de/westnordost/streetcomplete/StreetCompleteApplication.kt

```diff
@@ -155,6 +158,20 @@ class StreetCompleteApplication : Application() {
         applicationScope.cancel()
     }
 
+    override fun onTrimMemory(level: Int) {
+        super.onTrimMemory(level)
+        when (level) {
+            ComponentCallbacks2.TRIM_MEMORY_COMPLETE, ComponentCallbacks2.TRIM_MEMORY_RUNNING_CRITICAL -> {
+                // very low on memory -> drop caches
+                cacheTrimmer.clearCaches()
+            }
+            ComponentCallbacks2.TRIM_MEMORY_MODERATE, ComponentCallbacks2.TRIM_MEMORY_RUNNING_MODERATE, ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW -> {
```

Comment:
```suggestion
            ComponentCallbacks2.TRIM_MEMORY_COMPLETE, ComponentCallbacks2.TRIM_MEMORY_RUNNING_CRITICAL -> {
                // very low on memory -> drop caches
                cacheTrimmer.clearCaches()
            }
            ComponentCallbacks2.TRIM_MEMORY_MODERATE, ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW -> {
```

Reading through the docs, I think this is slightly better. No need to trim the cache only because the device **is beginning to run low on memory** and thus degrade the performance of this app **running in the foreground**. Other processes at the end of the LRU list should trim their memory first.